### PR TITLE
User compset fix

### LINKED ISSUE #1693: User-defined compsets differ in RUN_STARTDATE and other settings from config_compsets.xml
I think this is related to an issue @mvertens brought up verbally a few days ago: Any settings made in a config_compsets.xml file are not applied to user-compsets (i.e., compsets without an alias). One common variable that is set there is RUN_STARTDATE. This means, for example, that a user-defined HIST compset currently starts in year 0001 rather than year 1850, which I think leads to incorrect behavior.

I think the solution here is to either:

1. Avoid making any settings in config_compsets.xml other than settings that only apply to specific compsets. i.e., matches on a full compset long name are okay (as is done for setting up a refcase, for example), but more general matches are not allowed in this file and must go somewhere else. The driver's config_component seems like a good place for things like RUN_STARTDATE's dependence on the compset time period.

2. ( @mvertens 's tentative suggestion) Parse the primary component's config_compsets.xml file even when using a user-defined compset. Looking through the latest version of case.py, it looks like this could be done by moving this line:

```python
                    self.set_lookup_value("COMPSETS_SPEC_FILE" ,
                                   files.get_value("COMPSETS_SPEC_FILE", {"component":component}, resolved=False))
```

out of `_set_compset` and into `_set_info_from_primary_component`. To be safe, we should also remove the `_compsetsfile` variable from case.py (which is currently unused... if it were used, it would currently be inconsistent for user-defined vs aliased compsets).

It would be easy enough to make the change in (2); I'm just not sure what far-reaching implications it might have....

### PR DESCRIPTION
**Pull request by jedwards4b**
Rearrange code in case.py so that compset alias and compset longname get the same result. 

Test suite: scripts_regression_tests.py, test described in #1693 
Test baseline: 
Test namelist changes: 
Test status: bit for bit
Fixes #1693 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: scripts/lib/CIME/utils.py [hunk 1]
```diff
@@ -203,7 +203,8 @@ def get_model():
     # One last try
     if (model is None):
         srcroot = os.path.dirname(os.path.abspath(get_cime_root()))
-        if os.path.isfile(os.path.join(srcroot, "SVN_EXTERNAL_DIRECTORIES")):
+        if os.path.isfile(os.path.join(srcroot, "SVN_EXTERNAL_DIRECTORIES")) \
```
Comment (jedwards4b) [inline scripts/lib/CIME/utils.py]: This is an unrelated change so that the new cesm git repository works.

**Comment by jedwards4b** [comment]
@mnlevy1981 Sorry, please try again.

File: scripts/lib/CIME/case.py [hunk 2]
```diff
@@ -457,10 +456,8 @@ def _set_compset(self, compset_name, files):
                 compsets = Compsets(compsets_filename)
                 match, compset_alias, science_support = compsets.get_compset_match(name=compset_name)
                 if match is not None:
-                    self._compsetsfile = compsets_filename
                     self._compsetname = match
-                    self.set_lookup_value("COMPSETS_SPEC_FILE" ,
-                                   files.get_value("COMPSETS_SPEC_FILE", {"component":component}, resolved=False))
+                    self.set_lookup_value("COMPSETS_SPEC_FILE", compsets_filename)
```
[the remark this answers is not in the document]
Comment (jedwards4b) [reply scripts/lib/CIME/case.py]: Nope, I tried that - its needed in both places - perhaps you could write a test to boost your confidence?

[the remark this answers is not in the document]
Comment (jedwards4b) [reply scripts/lib/CIME/case.py]: It would need to be a case which is not dependent on an external component.   I think the
ADESP tests might do the trick.

[the remark this answers is not in the document]
Comment (billsacks) [reply scripts/lib/CIME/case.py]: Can you describe what doesn't work when that's removed? I just tried some manual tests with this diff:

```diff
diff --git a/scripts/lib/CIME/case.py b/scripts/lib/CIME/case.py
index b9a36ba72..07452e15f 100644
--- a/scripts/lib/CIME/case.py
+++ b/scripts/lib/CIME/case.py
@@ -457,7 +457,6 @@ def _set_compset(self, compset_name, files):
                 match, compset_alias, science_support = compsets.get_compset_match(name=compset_name)
                 if match is not None:
                     self._compsetname = match
-                    self.set_lookup_value("COMPSETS_SPEC_FILE", compsets_filename)
                     logger.info("Compset longname is {}".format(match))
                     logger.info("Compset specification file is {}".format(compsets_filename))
                     return compset_alias, science_support
```

I tried creating some cases with aliases and with longname (a compset that doesn't have an alias). My tests worked and gave the same final xml files as I got before introducing that change.

If it's true that that line is needed, then it seems that something is still wrong, because it implies that compsets with aliases are being treated differently in some way from compsets without aliases – which is what we want to avoid.

[the remark this answers is not in the document]
Comment (jedwards4b) [reply scripts/lib/CIME/case.py]: The problem was with compsets whose primary component was CPL.  I think that the 
PRE.f19_f19.ADESP_TEST.cheyenne_intel demonstrates.

**Comment by billsacks** [comment]
Given that this is needed in order to start alpha08g testing, I'm accepting it. I'd still like to understand the need for the line I've questioned, but this can be addressed in a separate PR.
